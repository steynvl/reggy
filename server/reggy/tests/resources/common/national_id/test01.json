{
  "type": "National ID",
  "information": {
    "kind": "Austrian social security number"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
