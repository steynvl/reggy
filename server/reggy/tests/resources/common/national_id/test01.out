Java
Pattern regex = Pattern.compile("\\d{4}(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}");
\\d{4}(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}
Perl
my $regex = /\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}/;
\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}
POSIX
\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}
\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}
Python
regex = re.compile(r'\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}')
\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}
JavaScript
const regex = /\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}/;
\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}
PHP
$regex = '/\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}/';
\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}
Golang
regex, _ := regexp.Compile("\\d{4}(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}")
\\d{4}(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}
Rust
let re = Regex::new(r"\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}").unwrap();
\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}
C#
var re = new Regex(@"\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}");
\d{4}(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}
Scala
val re = "\\d{4}(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}".r
\\d{4}(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}
Kotlin
val regex = Regex("\\d{4}(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}")
\\d{4}(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}